iothub/device/src/IotHubBaseClient.cs

```diff
@@ -156,6 +157,10 @@ public async Task SendEventAsync(Message message, CancellationToken cancellation
                 message.MessageId = Guid.NewGuid().ToString();
             }
 
+            message.PayloadConvention = _clientOptions.PayloadConvention;
+            message.ContentType = _clientOptions.PayloadConvention.PayloadSerializer.ContentType;
+            message.ContentEncoding = _clientOptions.PayloadConvention.PayloadEncoder.ContentEncoding.WebName;
```

Comment:
The outgoing message doesn't need the convention information until it reaches the transport layer. So, we could exclude this assignment and have the transport layer refer to the convention directly and retrieve the bytes.
The incoming message, however, does require the convention so that it can deserialize the payload and present it to the user. So, to keep them similar I've added the convention information into outgoing message as well.

However, they don't "need" to be similar. So, if we think that this processing in unnecessary, we can remove this logic altogether.